mllib-tests/v1p5/src/main/scala/mllib/perf/clustering/PICTest.scala

```diff
@@ -0,0 +1,62 @@
+package mllib.perf.clustering
+
+import org.json4s.JValue
+import org.json4s.JsonDSL._
+
+import org.apache.spark.SparkContext
+import org.apache.spark.graphx.{Edge, Graph}
+import org.apache.spark.mllib.clustering.PowerIterationClustering
+
+import mllib.perf.PerfTest
+
+class PICTest(sc: SparkContext) extends PerfTest {
+
+  val NUM_POINTS_SQUARED = ("num-points-squared", "number of points squared (for linear test-size scaling)")
+  val NODE_DEGREE = ("node-degree", "number of neighbors each node is connected to")
+  val NUM_CENTERS = ("num-centers", "number of centers for clustering tests")
+  val NUM_ITERATIONS = ("num-iterations", "number of iterations for the algorithm")
+
+  intOptions ++= Seq(NODE_DEGREE, NUM_CENTERS, NUM_ITERATIONS)
+  longOptions ++= Seq(NUM_POINTS_SQUARED)
+  val options = intOptions ++ stringOptions  ++ booleanOptions ++ longOptions ++ doubleOptions
+  addOptionsToParser()
+
+  var data: Graph[Double, Double] = _
+
+  override def createInputData(seed: Long): Unit = {
+    val numPoints = math.sqrt(longOptionValue(NUM_POINTS_SQUARED)).toInt
+    val nodeDegree = intOptionValue(NODE_DEGREE)
+    val numCenters = intOptionValue(NUM_CENTERS)
+    val numPartitions = intOptionValue(NUM_PARTITIONS)
+
+    // Generates a periodic banded matrix with bandwidth nodeDegree / 2
+    val similarities = sc.parallelize(0L to numPoints)
+      .flatMap { id =>
+        (((id - nodeDegree / 2) % numPoints) to ((id + nodeDegree / 2) % numPoints)).map { nbr =>
```

Comment:
`(id - nodeDegree / 2) % numPoints until id` to avoid duplicate. We should generate RDD, which is the main API, instead of Graph.